pkgs/development/python-modules/txtai/default.nix

```
@@ -176,23 +190,49 @@ buildPythonPackage {
 
   optional-dependencies = optional-dependencies;
 
-  pythonImportsCheck = [ "txtai" ];
-
-  # some tests hang forever
-  doCheck = false;
-
-  preCheck = ''
-    export TRANSFORMERS_CACHE=$(mktemp -d)
+  # The Python imports check runs huggingface-hub which needs a writable directory
+  postInstall = ''
```

```suggestion
  postInstall = ''
```
Usually, we would do that in `preCheck`.

---

It doesn't work in preCheck, unfortunately. Inspecting the build log shows that postCheck runs well before the python imports check. I spent quite a while finding the right place to set this value.

---

Here's what happens when I set this value in `postInstall`:

```sh
Running phase: installCheckPhase
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
Running phase: pythonRemoveBinBytecodePhase
Running phase: pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: txtai
Running phase: pytestCheckPhase
```

It works.

---

However, it works in `preInstall` which is in line with what the pore/post hooks should do.

---

Done as postInstall -> preInstall.